modules/core/src/encoder/SemanticPreEncoder.ts

```diff
@@ -127,6 +131,15 @@ export class SemanticPreEncoder {
 
             }
 
+            /**
+             * As special purposes have a special treatment and cannot being objected,
+             * these special purposes only vendors need to set LI signal when present
+             */
+
+            if (gvlVendorKey === 'legIntPurposes' && isSpecialPurposesOnly && !isInactive) {
+              vector.set(vendorId);
```

Comment:
We shouldn't unconditionally set the LI bit, as that would prevent the CMP from indicating "the vendor was NOT disclosed". We should set the bit ONLY if it was unset by lines #L71 or #L124 above, although it might be simpler to "avoid unsetting it" rather than "setting it back".

Reply:
agree!

Reply:
Hello @jcmlmorav, If you incorporate Andrew's @a2intl and Chris's @chrispaterson we'll review again and with hope approve the PR. Thank you.

Reply:
Sure, I will be working on this

Reply:
I opted to "avoid unsetting it" and remove extra lines defined before. (inactive variable as this is already covered in the first if)